base/applications/rapps/unattended.cpp

```diff
@@ -312,6 +313,48 @@ HandleInfoCommand(CAppDB *db, LPWSTR szCommand, int argcLeft, LPWSTR *argvLeft)
     return TRUE;
 }
 
+static unsigned int CALLBACK
+RunGuiThread(void *nCmdShow)
+{
+    WCHAR fakecmdline[] = L"://";
+    return ParseCmdAndExecute(fakecmdline, PCAEF_WINDOWONLY, (UINT)(SIZE_T)nCmdShow);
+}
+
+static BOOL
+HandleProtocolCommand(CAppDB &db, LPWSTR Url, int nCmdShow)
+{
+    if (Url[0] == '/' && Url[1] == '/')
+        Url += 2;
+
+    LPWSTR p;
+    if ((p = IsStrPrefixI(Url, L"install/")) != NULL)
+    {
+        UrlUnescapeInPlace(p, 0);
+        return HandleInstallCommand(&db, Url, 1, &p);
+    }
+
+    HANDLE hThread = (HANDLE)_beginthreadex(NULL, 0, RunGuiThread, (void*)(SIZE_T)nCmdShow, 0, NULL);
+    if (hThread)
+    {
+        HWND hWindow;
+        for (UINT timeout = 1000 * 5, interval = 250, i = 0; i < timeout; Sleep(interval), i += interval)
```

Comment:
Please move this Sleep() call inside the for-loop block instead. (So, after the break; below.)